source/MRMesh/MRCylinderApproximator.h

```diff
@@ -0,0 +1,413 @@
+#include "MRMeshFwd.h"
+#include "MRLine.h"
+#include "MRVector.h"
+#include "MRMatrix.h"
+#include <Eigen/Eigenvalues>
+#include "MRPch/MRTBB.h"
+#include "MRToFromEigen.h"
+#include "MRConstants.h"
+
+// https://www.geometrictools.com/Documentation/LeastSquaresFitting.pdf 
+
+namespace MR
+{
+
+//using T = float;
+template <typename T>
+class Cylinder3
+{
+public:
+    Cylinder3()
+    {}
+
+    Cylinder3( const Vector3<T>& inCenter, const Vector3<T>& inDirectoin, T inRadius, T inHeight )
+        :
+        mainAxis( inCenter, inDirectoin ),
+        radius( inRadius ),
+        length( inHeight )
+    {}
+    Cylinder3( const Line3<T>& inAxis, T inRadius, T inHeight )
+        :
+        mainAxis( inAxis ),
+        radius( inRadius ),
+        length( inHeight )
+    {}
+
+    inline MR::Vector3<T>& center( void )
+    {
+        return mainAxis.p;
+    }
+    inline MR::Vector3<T>& direction( void )
+    {
+        return mainAxis.d;
+    }
+
+    MR::Line3<T> mainAxis = {};
+    T radius = 0;
+    T length = 0;
+
+public:
+    bool operator==( Cylinder3 const& cylinder ) const = default;
+
+    bool operator!=( Cylinder3 const& cylinder ) const
+    {
+        return !operator==( cylinder );
+    }
+
+};
+
+//using T = float;  // very usefull for debug in VS.
+template <typename T>
+class Cylinder3DApproximation
```

Comment:
For consistency of namings, it shall be Cylinder3Approximation (not 3D).

Reply:
done